include/bluetooth/services/fast_pair.h

```diff
@@ -30,18 +37,76 @@ enum bt_fast_pair_adv_mode {
 
 	/** Fast Pair not discoverable advertising, hide UI indication. */
 	BT_FAST_PAIR_ADV_MODE_NOT_DISCOVERABLE_HIDE_UI_IND,
+
 	/** Number of Fast Pair advertising modes. */
 	BT_FAST_PAIR_ADV_MODE_COUNT
 };
 
+/** @brief Fast Pair advertising battery mode. Used to generate advertising packet.
+ *
+ * Battery data can be included in advertising packet only if the Fast Pair Provider is in Fast Pair
+ * not discoverable advertising mode. To prevent tracking, the Fast Pair Provider should not include
+ * battery data in the advertising packet all the time.
+ */
+enum bt_fast_pair_adv_battery_mode {
+	/** Do not advertise battery data. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_NONE,
+
+	/** Show battery data UI indication. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_SHOW_UI_IND,
+
+	/** Hide battery data UI indication. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_HIDE_UI_IND,
+
+	/** Number of Fast Pair advertising battery modes. */
+	BT_FAST_PAIR_ADV_BATTERY_MODE_COUNT
+};
+
+/** @brief Index of Fast Pair battery component. */
+enum bt_fast_pair_battery_comp {
+	/** Left bud. */
+	BT_FAST_PAIR_BATTERY_COMP_LEFT_BUD = 0,
+
+	/** Right bud. */
+	BT_FAST_PAIR_BATTERY_COMP_RIGHT_BUD = 1,
+
+	/** Case. */
+	BT_FAST_PAIR_BATTERY_COMP_BUD_CASE = 2,
+
+	/** Number of battery values. */
+	BT_FAST_PAIR_BATTERY_COMP_COUNT
+};
```

Comment:
I think we agreed to remove the battery infix to make component references general

```suggestion
/** @brief Index of Fast Pair component. */
enum bt_fast_pair_comp {
	/** Left bud. */
	BT_FAST_PAIR_COMP_LEFT_BUD = 0,

	/** Right bud. */
	BT_FAST_PAIR_COMP_RIGHT_BUD = 1,

	/** Case. */
	BT_FAST_PAIR_COMP_BUD_CASE = 2,

	/** Number of components. */
	BT_FAST_PAIR_COMP_COUNT
};
```

Also please move it below the ```bt_fast_pair_battery``` structure. It will follow an alphabetical order and will be near a structure that goes together with this one to the ```bt_fast_pair_battery_set``` API